File: 1063-best-sightseeing-pair/best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        res = 0
        m = 0
        for i in values:
            res = max(res, m + i)
            m = max(m - 1, i - 1)
        return res

File: 1063-best-sightseeing-pair/best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int result = 0;
        int m = 0;
        for (int i: values){
            result = Math.max(result, m + i);
            m = Math.max(m - 1, i - 1);
        }
        return result;
    }
}